filc/cmds.go

```diff
@@ -417,9 +409,37 @@ var retrieveFileCmd = &cli.Command{
 
 		// Do the retrieval
 
-		stats, err := node.RetrieveFromBestCandidate(cctx.Context, fc, c, selNode, candidates, CandidateSelectionConfig{
-			tryIPFS: !noIPFS,
-		})
+		var networks []interface{}
```

Comment:
why is this just an `interface{}`? seems like we could get the right level of abstraction here and not have to munge around with types

Reply:
well, this happened because `IPFSRetrievalConfig` and `FILRetrievalConfig` don't have any functions. really, this would be the ideal use case for a tagged union. if we want to make this more strict, i could put a dummy function on them to force them to a specific interface type.

Reply:
You could do something like:
```
type RetrieveOption interface {
  TryRetrieve(context.Context, node Node) (Stats, error)
}
```

Taking the try retrieve methods off of the node, and shifting them onto the config.

Alternatively, you could just do a 'simple' thing and give it a `Type` method (a dummy method, as you called it). This is less exciting, but still useful to avoid the `interface{}`

Reply:
you could also expose a `Priority` method that lets you sort the candidates by priority easily

Reply:
I like the RetrieveOption idea! I will implement it. The priority is already determined by the order they're passed into the retrieve function, and I want it to be configurable in that way, unless you meant something different!